## Practice Exercise 2

In this assignment, you will try to find some interesting insights into a few movies released between 1916 and 2016, using Python. You will have to download a movie dataset, write Python code to explore the data, gain insights into the movies, actors, directors, and collections, and submit the code.

#### Some tips before starting the assignment

1. Identify the task to be performed correctly, and only then proceed to write the required code. Don’t perform any incorrect analysis or look for information that isn’t required for the assignment.
2. In some cases, the variable names have already been assigned, and you just need to write code against them. In other cases, the names to be given are mentioned in the instructions. We strongly advise you to use the mentioned names only.
3. Always keep inspecting your data frame after you have performed a particular set of operations.
4. There are some checkpoints given in the IPython notebook provided. They're just useful pieces of information you can use to check if the result you have obtained after performing a particular task is correct or not.
5. Note that you will be asked to refer to documentation for solving some of the questions. That is done on purpose for you to learn new commands and also how to use the documentation.

In [33]:
# Import the numpy and pandas packages

import numpy as np
import pandas as pd

### Task 1: Reading and Inspection

**Subtask 1.1: Import and read**

Import and read the movie database. Store it in a variable called `movies`.

In [34]:
# Write your code for importing the csv file here
movies = pd.read_csv('Movies.csv')
movies.index.values

array([   0,    1,    2, ..., 3850, 3851, 3852], dtype=int64)

**Subtask 1.2: Inspect the dataframe**

Inspect the dataframe's columns, shapes, variable types etc.

In [21]:
# Write your code for inspection here


#### <font color='red'>Question 1: How many rows and columns are present in the dataframe? </font>
-  <font color='red'>(3821, 26)</font>
-  <font color='red'>(3879, 28)</font>
-  <font color='red'>(3853, 28)</font>
-  <font color='red'>(3866, 26)</font>

In [22]:
print(movies.shape)
movies.sum()

(3853, 28)


C:\Users\minak\AppData\Local\Temp/ipykernel_18704/742967281.py:2: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  movies.sum()


director_name                James CameronGore VerbinskiSam MendesChristoph...
num_critic_for_reviews                                                628057.0
duration                                                              423416.0
director_facebook_likes                                              3022707.0
actor_3_facebook_likes                                               2874825.0
actor_1_facebook_likes                                              29188627.0
gross                                                           196369968193.0
genres                       Action|Adventure|Fantasy|Sci-FiAction|Adventur...
actor_1_name                 CCH PounderJohnny DeppChristoph WaltzTom Hardy...
movie_title                  Avatar Pirates of the Caribbean: At World's En...
num_voted_users                                                      394639691
cast_total_facebook_likes                                             43305717
facenumber_in_poster                                

#### <font color='red'>Question 2: How many columns have null values present in them? Try writing a code for this instead of counting them manually.</font>

-  <font color='red'>3</font>
-  <font color='red'>6</font>
-  <font color='red'>9</font>
-  <font color='red'>12</font>

In [36]:
(movies.isna().sum()>0).sum()

12

### Task 2: Cleaning the Data

**Subtask 2.1: Drop unecessary columns**

For this assignment, you will mostly be analyzing the movies with respect to the ratings, gross collection, popularity of movies, etc. So many of the columns in this dataframe are not required. So it is advised to drop the following columns.
-  color
-  director_facebook_likes
-  actor_1_facebook_likes
-  actor_2_facebook_likes
-  actor_3_facebook_likes
-  actor_2_name
-  cast_total_facebook_likes
-  actor_3_name
-  duration
-  facenumber_in_poster
-  content_rating
-  country
-  movie_imdb_link
-  aspect_ratio
-  plot_keywords

In [38]:
# Check the 'drop' function in the Pandas library - dataframe.drop(list_of_unnecessary_columns, axis = )
# Write your code for dropping the columns here. It is advised to keep inspecting the dataframe after each set of operations

movies.drop(['color','director_facebook_likes','actor_1_facebook_likes','actor_2_facebook_likes','actor_3_facebook_likes','actor_2_name','cast_total_facebook_likes','actor_3_name','duration','facenumber_in_poster','content_rating','country','movie_imdb_link','aspect_ratio','plot_keywords'], axis=1, inplace=True)

In [39]:
movies.columns.size

13

#### <font color='red'>Question 3: What is the count of columns in the new dataframe? </font>
-  <font color='red'>10</font>
-  <font color='red'>13</font>
-  <font color='red'>15</font>
-  <font color='red'>17</font>

**Subtask 2.2: Inspect Null values**

As you have seen above, there are null values in multiple columns of the dataframe 'movies'. Find out the percentage of null values in each column of the dataframe 'movies'. 

In [40]:
# Write you code here
movies.isna().sum()

director_name             0
num_critic_for_reviews    1
gross                     0
genres                    0
actor_1_name              0
movie_title               0
num_voted_users           0
num_user_for_reviews      0
language                  3
budget                    0
title_year                0
imdb_score                0
movie_facebook_likes      0
dtype: int64

#### <font color='red'>Question 4: Which column has the highest percentage of null values? </font>
-  <font color='red'>language</font>
-  <font color='red'>genres</font>
-  <font color='red'>num_critic_for_reviews</font>
-  <font color='red'>imdb_score</font>

**Subtask 2.3: Fill NaN values**

You might notice that the `language` column has some NaN values. Here, on inspection, you will see that it is safe to replace all the missing values with `'English'`.

In [41]:
# Write your code for filling the NaN values in the 'language' column here
movies['language'] = movies['language'].fillna('English')

In [42]:
(movies['language'] == 'English').sum()

3674

#### <font color='red'>Question 5: What is the count of movies made in English language after replacing the NaN values with English? </font>
-  <font color='red'>3670</font>
-  <font color='red'>3674</font>
-  <font color='red'>3668</font>
-  <font color='red'>3672</font>

### Task 3: Data Analysis

**Subtask 3.1: Change the unit of columns**

Convert the unit of the `budget` and `gross` columns from `$` to `million $`.

In [43]:
# Write your code for unit conversion here
movies['budget'] = movies['budget'] / 1000000
movies['gross'] = movies['gross'] / 1000000
movies[['budget', 'gross']]

budget       gross
0     237.0000  760.505847
1     300.0000  309.404152
2     245.0000  200.074175
3     250.0000  448.130642
4     263.7000   73.058679
...        ...         ...
3848    0.0070    0.424760
3849    0.0070    0.070071
3850    0.0070    2.040920
3851    0.0090    0.004584
3852    0.0011    0.085222

[3853 rows x 2 columns]

**Subtask 3.2: Find the movies with highest profit**

   1. Create a new column called `profit` which contains the difference of the two columns: `gross` and `budget`.
   2. Sort the dataframe using the `profit` column as reference. (Find which command can be used here to sort entries from the documentation)
   3. Extract the top ten profiting movies in descending order and store them in a new dataframe - `top10`

In [44]:
# Write your code for creating the profit column here
movies['profit'] = movies['gross'] - movies['budget']
movies['profit']

0       523.505847
1         9.404152
2       -44.925825
3       198.130642
4      -190.641321
           ...    
3848      0.417760
3849      0.063071
3850      2.033920
3851     -0.004416
3852      0.084122
Name: profit, Length: 3853, dtype: float64

In [45]:
# Write your code for sorting the dataframe here
movies.sort_values('profit', ascending=False).head(10)

director_name  num_critic_for_reviews       gross  \
0         James Cameron                   723.0  760.505847   
28      Colin Trevorrow                   644.0  652.177271   
25        James Cameron                   315.0  658.672302   
2704       George Lucas                   282.0  460.935665   
2748   Steven Spielberg                   215.0  434.949459   
16          Joss Whedon                   703.0  623.279547   
482        Roger Allers                   186.0  422.783777   
230        George Lucas                   320.0  474.544677   
64    Christopher Nolan                   645.0  533.316061   
419           Gary Ross                   673.0  407.999255   

                                        genres         actor_1_name  \
0              Action|Adventure|Fantasy|Sci-Fi          CCH Pounder   
28            Action|Adventure|Sci-Fi|Thriller  Bryce Dallas Howard   
25                               Drama|Romance    Leonardo DiCaprio   
2704           Action|Adventure|Fantasy|Sci-Fi        Harrison Ford   
2748                             Family|Sci-Fi         Henry Thomas   
16                     Action|Adventure|Sci-Fi      Chris Hemsworth   
482   Adventure|Animation|Drama|Family|Musical    Matthew Broderick   
230            Action|Adventure|Fantasy|Sci-Fi      Natalie Portman   
64                 Action|Crime|Drama|Thriller       Christian Bale   
419            Adventure|Drama|Sci-Fi|Thriller    Jennifer Lawrence   

                                     movie_title  num_voted_users  \
0                                        Avatar            886204   
28                               Jurassic World            418214   
25                                      Titanic            793059   
2704         Star Wars: Episode IV - A New Hope            911097   
2748                 E.T. the Extra-Terrestrial            281842   
16                                 The Avengers            995415   
482                               The Lion King            644348   
230   Star Wars: Episode I - The Phantom Menace            534658   
64                              The Dark Knight           1676169   
419                            The Hunger Games            701607   

      num_user_for_reviews language  budget  title_year  imdb_score  \
0                   3054.0  English   237.0      2009.0         7.9   
28                  1290.0  English   150.0      2015.0         7.0   
25                  2528.0  English   200.0      1997.0         7.7   
2704                1470.0  English    11.0      1977.0         8.7   
2748                 515.0  English    10.5      1982.0         7.9   
16                  1722.0  English   220.0      2012.0         8.1   
482                  656.0  English    45.0      1994.0         8.5   
230                 3597.0  English   115.0      1999.0         6.5   
64                  4667.0  English   185.0      2008.0         9.0   
419                 1959.0  English    78.0      2012.0         7.3   

      movie_facebook_likes      profit  
0                    33000  523.505847  
28                  150000  502.177271  
25                   26000  458.672302  
2704                 33000  449.935665  
2748                 34000  424.449459  
16                  123000  403.279547  
482                  17000  377.783777  
230                  13000  359.544677  
64                   37000  348.316061  
419                 140000  329.999255

In [32]:
top10 = # Write your code to get the top 10 profiting movies here

SyntaxError: invalid syntax (Temp/ipykernel_18704/244981778.py, line 1)

**Checkpoint:** You might spot two movies directed by `James Cameron` in the list.

#### <font color='red'>Question 6: Which movie is ranked 5th from the top in the list obtained? </font>
-  <font color='red'>E.T. the Extra-Terrestrial</font>
-  <font color='red'>The Avengers</font>
-  <font color='red'>The Dark Knight</font>
-  <font color='red'>Titanic</font>

**Subtask 3.3: Find IMDb Top 250**

Create a new dataframe `IMDb_Top_250` and store the top 250 movies with the highest IMDb Rating (corresponding to the column: `imdb_score`). Also make sure that for all of these movies, the `num_voted_users` is greater than 25,000. 

Also add a `Rank` column containing the values 1 to 250 indicating the ranks of the corresponding films.

In [46]:
# Write your code for extracting the top 250 movies as per the IMDb score here. Make sure that you store it in a new dataframe 
# and name that dataframe as 'IMDb_Top_250'
IMDb_Top_250 = movies.sort_values('imdb_score', ascending=False).head(250)

#### <font color='red'>Question 7: Suppose movies are divided into 5 buckets based on the IMDb ratings: </font>
-  <font color='red'>7.5 to 8</font>
-  <font color='red'>8 to 8.5</font>
-  <font color='red'>8.5 to 9</font>
-  <font color='red'>9 to 9.5</font>
-  <font color='red'>9.5 to 10</font>

<font color = 'red'> Which bucket holds the maximum number of movies from *IMDb_Top_250*? </font>

In [47]:
IMDb_Top_250[(IMDb_Top_250.imdb_score >=7.5) & (IMDb_Top_250.imdb_score < 8)].shape

(35, 14)

In [48]:
IMDb_Top_250[(IMDb_Top_250.imdb_score >=8) & (IMDb_Top_250.imdb_score < 8.5)].shape

(168, 14)

**Subtask 3.4: Find the critic-favorite and audience-favorite actors**

   1. Create three new dataframes namely, `Meryl_Streep`, `Leo_Caprio`, and `Brad_Pitt` which contain the movies in which the actors: 'Meryl Streep', 'Leonardo DiCaprio', and 'Brad Pitt' are the lead actors. Use only the `actor_1_name` column for extraction. Also, make sure that you use the names 'Meryl Streep', 'Leonardo DiCaprio', and 'Brad Pitt' for the said extraction.
   2. Append the rows of all these dataframes and store them in a new dataframe named `Combined`.
   3. Group the combined dataframe using the `actor_1_name` column.
   4. Find the mean of the `num_critic_for_reviews` and `num_user_for_review` and identify the actors which have the highest mean.

In [49]:
Meryl_Streep = 
Leo_Caprio = IMDb_Top_250[IMDb_Top_250.actor_1_name == 'Leonardo DiCaprio']
Brad_Pitt = IMDb_Top_250[IMDb_Top_250.actor_1_name == 'Brad Pitt']

SyntaxError: invalid syntax (Temp/ipykernel_18704/3236433462.py, line 1)

In [50]:
# Write your code for creating three new dataframes here
Meryl_Streep = IMDb_Top_250[IMDb_Top_250.actor_1_name == 'Meryl Streep']

In [51]:
Leo_Caprio = IMDb_Top_250[IMDb_Top_250.actor_1_name == 'Leonardo DiCaprio']

In [52]:
Brad_Pitt = IMDb_Top_250[IMDb_Top_250.actor_1_name == 'Brad Pitt']

In [53]:
# Write your code for combining the three dataframes here
Combined = pd.concat([Meryl_Streep,Leo_Caprio,Brad_Pitt])
Combined

director_name  num_critic_for_reviews       gross  \
94        Christopher Nolan                   642.0  292.568851   
347         Martin Scorsese                   352.0  132.373442   
283       Quentin Tarantino                   765.0  162.804648   
294         Martin Scorsese                   606.0  116.866727   
173   Alejandro G. Iñárritu                   556.0  183.635922   
431         Martin Scorsese                   490.0  127.968405   
859        Steven Spielberg                   194.0  164.435221   
293            Edward Zwick                   166.0   57.366262   
646           David Fincher                   315.0   37.023395   
2602             Tony Scott                   122.0   12.281500   

                                   genres       actor_1_name  \
94       Action|Adventure|Sci-Fi|Thriller  Leonardo DiCaprio   
347                  Crime|Drama|Thriller  Leonardo DiCaprio   
283                         Drama|Western  Leonardo DiCaprio   
294          Biography|Comedy|Crime|Drama  Leonardo DiCaprio   
173      Adventure|Drama|Thriller|Western  Leonardo DiCaprio   
431                      Mystery|Thriller  Leonardo DiCaprio   
859                 Biography|Crime|Drama  Leonardo DiCaprio   
293              Adventure|Drama|Thriller  Leonardo DiCaprio   
646                                 Drama          Brad Pitt   
2602  Action|Crime|Drama|Romance|Thriller          Brad Pitt   

                   movie_title  num_voted_users  num_user_for_reviews  \
94                  Inception           1468200                2803.0   
347              The Departed            873649                2054.0   
283          Django Unchained            955174                1193.0   
294   The Wolf of Wall Street            780588                1138.0   
173              The Revenant            406020                1188.0   
431            Shutter Island            786092                 964.0   
859       Catch Me If You Can            525801                 667.0   
293             Blood Diamond            400292                 657.0   
646                Fight Club           1347461                2968.0   
2602             True Romance            163492                 460.0   

     language  budget  title_year  imdb_score  movie_facebook_likes  \
94    English   160.0      2010.0         8.8                175000   
347   English    90.0      2006.0         8.5                 29000   
283   English   100.0      2012.0         8.5                199000   
294   English   100.0      2013.0         8.2                138000   
173   English   135.0      2015.0         8.1                190000   
431   English    80.0      2010.0         8.1                 53000   
859   English    52.0      2002.0         8.0                 15000   
293   English   100.0      2006.0         8.0                 14000   
646   English    63.0      1999.0         8.8                 48000   
2602  English    13.0      1993.0         8.0                 15000   

          profit  
94    132.568851  
347    42.373442  
283    62.804648  
294    16.866727  
173    48.635922  
431    47.968405  
859   112.435221  
293   -42.633738  
646   -25.976605  
2602   -0.718500

In [57]:
# Write your code for grouping the combined dataframe here
comb_group = Combined.groupby(['actor_1_name'])
comb_group.mean().loc['Brad Pitt']

num_critic_for_reviews       218.500000
gross                         24.652448
num_voted_users           755476.500000
num_user_for_reviews        1714.000000
budget                        38.000000
title_year                  1996.000000
imdb_score                     8.400000
movie_facebook_likes       31500.000000
profit                       -13.347552
Name: Brad Pitt, dtype: float64

In [ ]:
# Write the code for finding the mean of critic reviews and audience reviews here
comb_group.mean()[['num_critic_for_reviews', 'num_voted_users']]

#### <font color='red'>Question 8: Which actor is highest rated among the three actors according to the user reviews? </font>
-  <font color='red'>Meryl Streep</font>
-  <font color='red'>Leonardo DiCaprio</font>
-  <font color='red'>Brad Pitt</font>

#### <font color='red'>Question 9: Which actor is highest rated among the three actors according to the critics?</font>
-  <font color='red'>Meryl Streep</font>
-  <font color='red'>Leonardo DiCaprio</font>
-  <font color='red'>Brad Pitt</font>

In [ ]:
IMDb_Top_250.columns = [col.capitalize() for col in IMDb_Top_250.columns]
IMDb_Top_250.columns

In [65]:
movies_final = pd.read_csv('movies_final (2).csv', sep='\t')
print(movies_final.columns)
movies_final = movies_final.loc[:,['actor_1_name','actor_1_facebook_likes','actor_2_name', 'actor_2_facebook_likes', 'actor_3_name', 'actor_3_facebook_likes']]

Index(['Unnamed: 0', 'color', 'director_name', 'num_critic_for_reviews',
       'duration', 'director_facebook_likes', 'actor_3_facebook_likes',
       'actor_2_name', 'actor_1_facebook_likes', 'gross', 'genres',
       'actor_1_name', 'movie_title', 'num_voted_users',
       'cast_total_facebook_likes', 'actor_3_name', 'facenumber_in_poster',
       'plot_keywords', 'movie_imdb_link', 'num_user_for_reviews', 'language',
       'country', 'content_rating', 'budget', 'title_year',
       'actor_2_facebook_likes', 'imdb_score', 'aspect_ratio',
       'movie_facebook_likes'],
      dtype='object')


In [66]:
movies_final['total_votes'] = movies_final['actor_1_facebook_likes'] + movies_final['actor_2_facebook_likes']
+ movies_final['actor_3_facebook_likes']

movies_final.sort_values('total_votes', ascending=False, inplace=True)
print(movies_final.head())

         actor_1_name  actor_1_facebook_likes     actor_2_name  \
1766    Darcy Donavan                640000.0     Will Ferrell   
1155     Krista Allen                164000.0  Andrew Fiscella   
3601     Matthew Ziff                260000.0      Randy Wayne   
2641  Andrew Fiscella                137000.0    Jay Hernandez   
2086  Andrew Fiscella                137000.0        Fred Ward   

      actor_2_facebook_likes           actor_3_name  actor_3_facebook_likes  \
1766                  8000.0           Steve Carell                  7000.0   
1155                137000.0      Shantel VanSanten                   748.0   
3601                   984.0  Raquel Elizabeth Ames                   891.0   
2641                  1000.0          Dania Ramirez                  1000.0   
2086                   459.0            Lorna Raver                   163.0   

      total_votes  
1766     648000.0  
1155     301000.0  
3601     260984.0  
2641     138000.0  
2086     137459.0  


In [78]:
movies_final[((2*(min(movies_final['actor_3_facebook_likes'], movies_final['actor_2_facebook_likes'], 
        movies_final['actor_1_facebook_likes']))) > (max(movies_final['actor_3_facebook_likes'], movies_final['actor_2_facebook_likes'], 
        movies_final['actor_1_facebook_likes'])))]

ValueError: The truth value of a Series is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().

In [1]:
import pandas as pd
movies_final = pd.read_csv('movies_final (2).csv', sep='\t')

In [3]:
movies_final[movies_final['actor_1_facebook_likes'].isna()]

Empty DataFrame
Columns: [Unnamed: 0, color, director_name, num_critic_for_reviews, duration, director_facebook_likes, actor_3_facebook_likes, actor_2_name, actor_1_facebook_likes, gross, genres, actor_1_name, movie_title, num_voted_users, cast_total_facebook_likes, actor_3_name, facenumber_in_poster, plot_keywords, movie_imdb_link, num_user_for_reviews, language, country, content_rating, budget, title_year, actor_2_facebook_likes, imdb_score, aspect_ratio, movie_facebook_likes]
Index: []

[0 rows x 29 columns]

In [5]:
# Importing the pandas package
import pandas as pd

# Reading the movies file
movies = pd.read_csv('https://media-doselect.s3.amazonaws.com/generic/pLLQowB8OYx0oBWdMbY4gp4wb/movies_final (2).csv', sep='\t')

# Group the dataframe using the actor names as indices and facebook likes as values
group = movies.pivot_table(values = ['actor_1_facebook_likes', 'actor_2_facebook_likes', 'actor_3_facebook_likes'],
                           index = ['actor_1_name', 'actor_2_name', 'actor_3_name'], aggfunc='sum')
                           


actor_1_facebook_likes  \
actor_1_name    actor_2_name    actor_3_name                                  
50 Cent         Bill Duke       Marc John Jefferies                  1000.0   
Aaliyah         Lena Olin       Bruce Spence                          775.0   
Aasif Mandvi    Dequina Moore   Jordan Carlos                         346.0   
Abbie Cornish   Gabourey Sidibe Michael Stuhlbarg                    2000.0   
                Paul Schneider  Samuel Roukin                        2000.0   
...                                                                     ...   
Zooey Deschanel Rip Torn        Kelly Preston                       11000.0   
                Will Ferrell    Dallas Roberts                      11000.0   
Zoë Kravitz     Jeremy Davies   Jim Gaffigan                          943.0   
                Lily Rabe       Bob Balaban                           943.0   
Óscar Jaenada   Brett Cullen    Sedona Legge                          619.0   

                                                     actor_2_facebook_likes  \
actor_1_name    actor_2_name    actor_3_name                                  
50 Cent         Bill Duke       Marc John Jefferies                  1000.0   
Aaliyah         Lena Olin       Bruce Spence                          541.0   
Aasif Mandvi    Dequina Moore   Jordan Carlos                          41.0   
Abbie Cornish   Gabourey Sidibe Michael Stuhlbarg                     906.0   
                Paul Schneider  Samuel Roukin                         552.0   
...                                                                     ...   
Zooey Deschanel Rip Torn        Kelly Preston                         826.0   
                Will Ferrell    Dallas Roberts                       8000.0   
Zoë Kravitz     Jeremy Davies   Jim Gaffigan                          769.0   
                Lily Rabe       Bob Balaban                           763.0   
Óscar Jaenada   Brett Cullen    Sedona Legge                          350.0   

                                                     actor_3_facebook_likes  
actor_1_name    actor_2_name    actor_3_name                                 
50 Cent         Bill Duke       Marc John Jefferies                   441.0  
Aaliyah         Lena Olin       Bruce Spence                          531.0  
Aasif Mandvi    Dequina Moore   Jordan Carlos                          31.0  
Abbie Cornish   Gabourey Sidibe Michael Stuhlbarg                     816.0  
                Paul Schneider  Samuel Roukin                         179.0  
...                                                                     ...  
Zooey Deschanel Rip Torn        Kelly Preston                         742.0  
                Will Ferrell    Dallas Roberts                        405.0  
Zoë Kravitz     Jeremy Davies   Jim Gaffigan                          472.0  
                Lily Rabe       Bob Balaban                           559.0  
Óscar Jaenada   Brett Cullen    Sedona Legge                            2.0  

[3631 rows x 3 columns]

In [6]:
# Create a new column 'Total likes' which will contain the sum of likes of all three actors 
group['Total likes'] = group['actor_1_facebook_likes'] + group['actor_2_facebook_likes'] + group['actor_3_facebook_likes']

In [7]:
# Sort the dataframe using the 'Total likes' column
group.sort_values(by=['Total likes'], inplace=True, ascending = False)

In [11]:
# Reset the index of the grouped dataframe so you can access the indices as columns easily
group.reset_index(inplace=True)
group['Total likes']

0       655000.0
1       301748.0
2       261875.0
3       228000.0
4       198000.0
          ...   
3626         2.0
3627         2.0
3628         2.0
3629         2.0
3630         0.0
Name: Total likes, Length: 3631, dtype: float64

In [9]:
# Initialise the value of a variable 'j' to 0. This variable will be used to keep
# a track of the rows during the loop iteration
j = 0

# Run a loop through the length of the column 'Total likes'
for i in group['Total likes']:
# Sort the facebook likes of three actors and store them in a variable 'temp'    
    temp = sorted([group.loc[j,'actor_1_facebook_likes'], group.loc[j,'actor_2_facebook_likes'], group.loc[j,'actor_3_facebook_likes']])

# Check if the smallest value in temp is greater than half the value of the other two
# And also check if the middle value is greater than half the value of the max value
    if temp[0] >= temp[1]/2 and temp[0] >= temp[2]/2 and temp[1] >= temp[2]/2:
# If the above condition satisfies, print the correspoding actor names as a sorted list 
# and break the loop
        print(sorted([group.loc[j, 'actor_1_name'], group.loc[j, 'actor_2_name'], group.loc[j, 'actor_3_name']]))
        break
# Keep incrementing the value of j with every loop interation    
    j += 1 # Initialise the value of a variable 'j' to 0. This variable will be used to keep
# a track of the rows during the loop iteration
j = 0

# Run a loop through the length of the column 'Total likes'
for i in group['Total likes']:
# Sort the facebook likes of three actors and store them in a variable 'temp'    
    temp = sorted([group.loc[j,'actor_1_facebook_likes'], group.loc[j,'actor_2_facebook_likes'], group.loc[j,'actor_3_facebook_likes']])

# Check if the smallest value in temp is greater than half the value of the other two
# And also check if the middle value is greater than half the value of the max value
    if temp[0] >= temp[1]/2 and temp[0] >= temp[2]/2 and temp[1] >= temp[2]/2:
# If the above condition satisfies, print the correspoding actor names as a sorted list 
# and break the loop
        print(sorted([group.loc[j, 'actor_1_name'], group.loc[j, 'actor_2_name'], group.loc[j, 'actor_3_name']]))
        break
# Keep incrementing the value of j with every loop interation    
    j += 1 

['Chris Hemsworth', 'Robert Downey Jr.', 'Scarlett Johansson']
['Chris Hemsworth', 'Robert Downey Jr.', 'Scarlett Johansson']
